### ISSUE TITLE
Offer "Copy link to topic"

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#792]
We should offer a "Copy link to topic" option.

This will be part of the topic action sheet. At the moment we don't have a topic action sheet; this can be the first item in it, or this issue can come after one of the other open issues for options to add to the topic action sheet:
* #348 
* #744

In particular see #348 for a description of where the topic action sheet should appear.

When we do offer this option, the links it generates should be topic permalinks, when the server supports it (it's part of the upcoming Zulip Server 9.0) and once we're ready for Zulip clients in general to be generating topic permalinks. In other words, the zulip-flutter implementation of this feature should match this zulip-mobile issue:
* https://github.com/zulip/zulip-mobile/issues/5876

and the eventual version (i.e. the version that gets merged, after any revisions) of the web side of this PR:
* https://github.com/zulip/zulip/pull/30644

**Comment by lakshya1goel** [comment]
Hi, I’m interested in working on the "Copy link to topic" feature for the topic action sheet. Before proceeding with the UI and integration, I wanted to confirm whether Zulip Server 9.0, which supports topic permalinks, is already available or if there’s a specific timeline for its release.

Since the feature involves generating topic links as permalinks, I’d like to ensure the server is ready to handle this before implementing it on the Flutter side.

Looking forward to any updates!
Thanks!

**Comment by gnprice** [comment]
Yes, if you do a web search for "zulip server 9.0", you'll find the announcement:
https://blog.zulip.com/2024/07/25/zulip-9-0-released/

In any case a PR for this issue will need to support generating both permalinks and the older non-permalinks. As I said above, we'll want to generate permalinks only when the server supports it; and not all servers have upgraded to 9.0. See our README at https://github.com/zulip/zulip-flutter#server-compatibility, and see the existing code for examples of how we handle differences between server versions.

**Comment by chrisbobbe** [comment]
When this is done we should update the Help Center doc. We removed instructions for the legacy app when preparing to launch the new app: zulip/zulip@3a626036d1d2da8c0955516f43b4a90ea46a957e

**Comment by gnprice** [comment]
This was requested yesterday [in the Rust community](https://rust-lang.zulipchat.com/#narrow/channel/122653-zulip/topic/new.20mobile.20app.20feedback/near/527993981).
